packages/engine/src/datastore/table/proxy.rs

```diff
@@ -72,7 +73,8 @@ impl<K: Batch> Clone for BatchReadProxy<K> {
 
 impl<K: Batch> Drop for BatchReadProxy<K> {
     fn drop(&mut self) {
-        unsafe { self.arc.raw().unlock_shared() }
+        // SAFETY: We are holding a shared lock, so we can unlock it
```

Comment:
I don't quite understand this. Is it releasing this specific shared lock? 
Or is it releasing all shared locks?

Reply:
It's releasing `self`'s lock only. I clarified the comment.